pfio/cache/multiprocess_file_cache.py

```diff
@@ -349,8 +350,9 @@ def preserve(self, name):
         data_file = os.path.join(self.dir, '{}.cached'.format(name))
 
         if any(os.path.exists(p) for p in (index_file, data_file)):
-            raise ValueError('Specified cache name "{}" already exists in {}'
-                             .format(name, self.dir))
+            msg = 'Specified cache named "{}" already exists in {}' \
```

Comment:
I wonder whether we should raise an error in this case. IMO, user might run the same script for multiple times with different parameters, but the cache file name will be the same. If we raise an error here, the job will fail when they do the second run, and user has to remember to clean up the cache each time.

Since it is a read-only cache, do you see any issue if we overwrite it if the same are the same?

Reply:
In this PR I intended to have the identical behavior of the original `FileCache` ([os.link called here](https://github.com/pfnet/pfio/blob/1.0.0/pfio/cache/file_cache.py#L303) will raise a `FileExistsError`. Similarly for `preload`, btw).
But I indeed have had the similar concerning to yours. At that moment I considered implementing an 'overwrite=True' option but "always-overwrite" approach looks more handy to me. If you don't mind changing the current behavior I'm happy to update the PR :+1:

Reply:
This might be related; I have thought of changing the behavior of `preload` when no cache with the specified name exists.
Currently it simply raises `FileNotFoundError` (following the `FileCache` behavior), but similarly to your concerning on repetitive run of a script, it'd be handy if it does nothing when no cache data exists yet.
If a user wants to realize the same thing, currently he/she need to manually check the existence of cache data (which requires several path string manipulations) and call `preload` accordingly.

Reply:
@belldandyxtq
It's been a while, but I modified the interface of `FileCache` and `MultiprocessFileCache`.
Specifically `preserve` and `preload` return `True` if they succeed and `False` otherwise. Fatal errors such as OSError or invalid use (call preserve/preload in child process) will still raise an Exception, but `FileNotFound` will no longer be thrown.